modules/channel-telegram/src/backend/client.ts

```diff
@@ -10,24 +10,30 @@ const outgoingTypes = ['text', 'typing', 'image', 'login_prompt', 'carousel']
 
 export const sendEvent = async (bp: typeof sdk, botId: string, ctx: ContextMessageUpdate, args: { type: string }) => {
   // NOTE: getUpdate and setWebhook dot not return the same context mapping
-  const threadId = _.get(ctx, 'chat.id') || _.get(ctx, 'message.chat.id')
-  const target = _.get(ctx, 'from.id') || _.get(ctx, 'message.from.id')
-
-  const payload = _.get(ctx, 'message') || _.get(ctx, 'callback_query')
-  const preview = _.get(ctx, 'message.text') || _.get(ctx, 'callback_query.data')
-
-  await bp.events.sendEvent(
-    bp.IO.Event({
-      botId,
-      payload,
-      preview,
-      channel: 'telegram',
-      direction: 'incoming',
-      threadId: threadId && threadId.toString(),
-      target: target && target.toString(),
-      ...args
-    })
-  )
+  const chatId = `${ctx.chat?.id || ctx.message?.chat.id}`
+  const userId = `${ctx.from?.id || ctx.message?.from.id}`
```

Comment:
I think you should also validate that the userId is not empty and raise an error if it's the case!